### PR TITLE
Bug/1700 assets manifest

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #1700  

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Core/Assets/Assets.php [hunk 1]
```diff
@@ -347,7 +347,7 @@ private function get_assets() {
 		$dependencies = array(
 			'googlesitekit-vendor',
 			'googlesitekit-commons',
-			'googlesitekit-base',
+			'googlesitekit-admin',
```
Comment (felixarntz) [inline includes/Core/Assets/Assets.php:350]: This shouldn't be changed to `admin`. Rather, let's rename the actual file `googlesitekit-admin.js` to `googlesitekit-base.js` which was the original plan here.

File: includes/Core/Assets/Assets.php [hunk 2]
```diff
@@ -373,7 +373,7 @@ private function get_assets() {
 				)
 			),
 			new Script_Data(
-				'googlesitekit-base-data',
+				'googlesitekit-admin-data',
```
Comment (felixarntz) [inline includes/Core/Assets/Assets.php:376]: See above, this should remain `googlesitekit-base-data`.

File: includes/Core/Assets/Assets.php [hunk 3]
```diff
@@ -433,23 +433,23 @@ private function get_assets() {
 			new Script(
 				'googlesitekit-activation',
 				array(
-					'src'          => $base_url . 'js/googlesitekit-activation.js',
+					'version'      => null,
```
Comment (felixarntz) [inline includes/Core/Assets/Assets.php:436]: Instead of setting `version` to `null` here throughout the file, let's handle this in the `Script` class automatically: If a file is in the manifest, no need to specify the version. If it is not in the manifest, we would specify the version.

File: includes/Core/Assets/Assets.php [hunk 4]
```diff
@@ -515,51 +515,51 @@ private function get_assets() {
...
 					'dependencies' => array(
 						'googlesitekit-data',
 					),
 				)
 			),
 			// End JSR Assets.
 			new Script(
-				'googlesitekit-ads-detect',
+				'pagead2.ads',
```
Comment (felixarntz) [inline includes/Core/Assets/Assets.php:526]: Let's keep the `googlesitekit-` prefix here, the handle should still be scoped to Site Kit. What we can do instead to make it work is:
* Either add a special case for this one script hard-coded to the `Script` class.
* Or, if the handle does not exist in the manifest, strip the `googlesitekit-` prefix and try lookup again.

File: includes/Core/Assets/Script.php [hunk 5]
```diff
@@ -53,15 +55,23 @@ public function __construct( $handle, array $args ) {
...
 	 */
-	public function register() {
+	public function register( Context $context ) {
 		if ( $this->args['fallback'] && wp_script_is( $this->handle, 'registered' ) ) {
 			return;
 		}
 
+		$src = $this->args['src'];
+		if ( class_exists( '\Google\Site_Kit\Core\Assets\Manifest' ) && isset( Manifest::$assets[ $this->get_handle() ] ) ) {
+			$src = $context->url( 'dist/assets/js/' . Manifest::$assets[ $this->get_handle() ] );
```
Comment (felixarntz) [inline includes/Core/Assets/Script.php:69]: Based on my previous comment: If this condition is fulfilled, let's automatically specify `version` as `null` in the `wp_register_script` call below.

File: includes/Core/Assets/Assets.php [hunk 6]
```diff
@@ -433,23 +433,23 @@ private function get_assets() {
 			new Script(
 				'googlesitekit-activation',
 				array(
-					'src'          => $base_url . 'js/googlesitekit-activation.js',
```
Comment (felixarntz) [inline includes/Core/Assets/Assets.php:436]: We should keep these `src` arguments around I'd say. It will be overridden internally by the `Script` class, and having them here makes it more obvious what file it actually is.

File: includes/Modules/AdSense.php [hunk 7]
```diff
@@ -857,13 +857,11 @@ protected function setup_settings() {
...
 	 */
 	protected function setup_assets() {
-		$base_url = $this->context->url( 'dist/assets/' );
-
 		return array(
 			new Script(
 				'googlesitekit-modules-adsense',
 				array(
-					'src'          => $base_url . 'js/googlesitekit-modules-adsense.js',
+					'version'      => null,
```
Comment (felixarntz) [inline includes/Modules/AdSense.php:864]: See other comment, let's keep `src` in here and omit the `version` argument here. This can all be handled internally in the `Script` class.

File: tests/phpunit/integration/Core/Assets/AssetsTest.php [hunk 8]
```diff
@@ -88,15 +88,15 @@ public function test_register() {
...
 
 		$assets = new Assets( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
 
-		$assets->enqueue_asset( 'googlesitekit-base' );
+		$assets->enqueue_asset( 'googlesitekit-admin' );
 
-		$this->assertTrue( wp_script_is( 'googlesitekit-base', 'registered' ) );
-		$this->assertTrue( wp_script_is( 'googlesitekit-base', 'enqueued' ) );
+		$this->assertTrue( wp_script_is( 'googlesitekit-admin', 'registered' ) );
+		$this->assertTrue( wp_script_is( 'googlesitekit-admin', 'enqueued' ) );
```
Comment (felixarntz) [inline tests/phpunit/integration/Core/Assets/AssetsTest.php:99]: See above, these should be reverted to using `base`.

File: tests/e2e/specs/modules/analytics/setup-proxy-no-account-no-tag.test.js [hunk 9]
```diff
@@ -75,7 +75,7 @@ describe( 'setting up the Analytics module with no existing account and no exist
 		await expect( page ).toClick( '.googlesitekit-analytics__select-country' );
 		await expect( page ).toClick( '.mdc-menu-surface--open li', { text: /united kingdom/i } );
 
-		await expect( page ).toMatchElement( 'p', { text: /need to give Site Kit permission to create an Analytics account/i } );
+		await expect( page ).toMatchElement( 'p', { text: /need to give Site Kit permission to create an Analytics account/i, timeout: 750 } );
```
Comment (felixarntz) [inline tests/e2e/specs/modules/analytics/setup-proxy-no-account-no-tag.test.js:78]: That seems unrelated - did that fix a random test failure you noticed while working on this issue?

File: includes/Core/Assets/Script.php [hunk 10]
```diff
@@ -53,15 +55,23 @@ public function __construct( $handle, array $args ) {
...
+	 * @param Context $context Plugin context.
 	 */
-	public function register() {
+	public function register( Context $context ) {
 		if ( $this->args['fallback'] && wp_script_is( $this->handle, 'registered' ) ) {
 			return;
 		}
 
+		$src = $this->args['src'];
+		if ( class_exists( '\Google\Site_Kit\Core\Assets\Manifest' ) && isset( Manifest::$assets[ $this->get_handle() ] ) ) {
```
Comment (felixarntz) [inline includes/Core/Assets/Script.php:68]: We should probably expect this class to exist here, otherwise our plugin is broken.
```suggestion
		if ( isset( Manifest::$assets[ $this->handle ] ) ) {
```

Also, based on my other comment, if this is not found, let's try stripping the `googlesitekit-` prefix and finding it from that.

File: includes/Core/Assets/Script.php (hunk 5 above)
Comment (felixarntz) [inline includes/Core/Assets/Script.php:69]:
```suggestion
			$src = $context->url( 'dist/assets/js/' . Manifest::$assets[ $this->handle ] );
```

File: includes/Core/Assets/Assets.php (hunk 1 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Assets.php]: Updated.

File: includes/Core/Assets/Assets.php (hunk 2 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Assets.php]: Updated.

File: includes/Core/Assets/Assets.php (hunk 3 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Assets.php]: Updated.

File: includes/Core/Assets/Script.php (hunk 5 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Script.php]: Updated.

File: includes/Modules/AdSense.php (hunk 7 above)
Comment (eugene-manuilov) [reply includes/Modules/AdSense.php]: Updated.

File: tests/phpunit/integration/Core/Assets/AssetsTest.php (hunk 8 above)
Comment (eugene-manuilov) [reply tests/phpunit/integration/Core/Assets/AssetsTest.php]: Updated.

File: includes/Core/Assets/Script.php (hunk 10 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Script.php]: Updated.

File: includes/Core/Assets/Script.php (hunk 5 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Script.php]: Updated.

File: includes/Core/Assets/Assets.php (hunk 6 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Assets.php]: Reverted.

File: includes/Core/Assets/Assets.php (hunk 4 above)
Comment (eugene-manuilov) [reply includes/Core/Assets/Assets.php]: Updated.

File: tests/e2e/specs/modules/analytics/setup-proxy-no-account-no-tag.test.js (hunk 9 above)
Comment (eugene-manuilov) [reply tests/e2e/specs/modules/analytics/setup-proxy-no-account-no-tag.test.js]: Yeah, that helped with a weird issue, but now it seems to be not needed anymore. Reverted.

**Comment by eugene-manuilov** [comment]
@felixarntz updated.

File: .travis.yml [hunk 11]
```diff
@@ -83,6 +83,7 @@ script:
   - |
     if [[ "$E2E" == "1" ]]; then
       npm run build:test || exit 1 # Build for tests.
+      docker run --rm -v "$PWD:/app" -v "$HOME/.cache/composer:/tmp/cache" composer install
```
Comment (aaemnnosttv) [inline .travis.yml:86]: @eugene-manuilov Why was this added when the same command is already run for E2E tests above on line `47` in the `before_script`?

Comment (eugene-manuilov) [reply .travis.yml:86]: We need to regenerate autoload map to include Manifest.php file generated after `npm run build:test`. Probably not the best approach to do it, but it works.

Comment (aaemnnosttv) [reply .travis.yml:86]: Yeah, worst case we can run `composer run autoload-includes` to only regenerate the Site Kit includes without needing to run all of the prefixing again, but I think we can do better than that.
